mitodl/open-discussions
# new open API part 2: withcertificate

#### Pre-Flight checklist


#### What are the relevant tickets?
updates https://github.com/mitodl/open-discussions/issues/3868

#### What's this PR do?
This PR adds  a new filter to the filterset for courses: withcertificate

The certificated filter is based on the [logic from the certification property](https://github.com/mitodl/open-discussions/blob/master/course_catalog/models.py#L320).

#### How should this be manually tested?
populate data using the backpopulate management commands. I would suggest mitx, a professional, and an open platform at the very least to test all three branches

```manage.py recreate_index --courses
manage.py backpopulate_mitxonline_data
manage.py backpopulate_mitxonline_files
manage.py backpopulate_xpro_data
manage.py backpopulate_xpro_files
manage.py backpopulate_ocw_data
manage.py backpopulate_ocw_files
```

Once those complete, you can use your method of choice to hit the following URLS:
`http://od.odl.local:8063/api/v0/courses/?withcertificate =True` - should show courses that show "Certificates" in the certification field
`http://od.odl.local:8063/api/v0/courses/?withcertificate =False` - should not have any courses that show "Certificates" in the certification field
`http://od.odl.local:8063/api/v0/courses/` - should show all courses

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
There's a slight addition to another test to cover a missing case
